cache/cache.go

```diff
@@ -331,7 +331,7 @@ func (in *kialiCacheImpl) GetZtunnelPods(cluster string) []v1.Pod {
 	selector := map[string]string{
 		config.IstioAppLabel: config.Ztunnel,
 	}
-	daemonsets, err := kubernetes.FilterDaemonSetsBySelector(daemonSetList.Items, selector)
+	daemonsets, err := kubernetes.FilterDaemonSetsBySelector(daemonSetList.Items, selector, true)
```

Comment:
Can we get rid of `kubernetes.FilterDaemonSetsBySelector` entirely and instead pass this label: https://github.com/istio/istio/blob/25e8af981fdc1730a7fa1127d3db4a8b2e54052f/manifests/charts/ztunnel/templates/daemonset.yaml#L7 as a list option to `kubeCache.List(...)`?

Reply:
Need some unit tests covering the case where the ztunnel has this label and some variation of selector labels. If we use the new metadata.label the selector labels shouldn't matter anymore.

Reply:
I like the idea of removing `kubernetes.FilterDaemonSetsBySelector` but if we are looking for ztunnel workloads then I think maybe we should continue to use the label selector approach and use something like

listOpts := []client.ListOption{client.MatchingLabels(selector)}

If we don't care about workloads and just want to look for the DS, then what Nick suggests seems reasonable.

Also, look at other instances of `FilterXXXBySelector`, it looks like we have other places where those filter calls follow a call to `kubeCache.List`, and the selector could be applied to the list options.